packages/wallets/src/components/Base/ModalWrapper/ModalWrapper.tsx

```diff
@@ -18,22 +19,45 @@ const ModalWrapper: FC<React.PropsWithChildren<TProps>> = ({
     shouldPreventCloseOnEscape = false,
 }) => {
     const { hide } = useModal();
+    const { isMobile } = useDevice();
 
     useEventListener('keydown', (event: KeyboardEvent) => {
         if (!shouldPreventCloseOnEscape && event.key === 'Escape') {
             hide();
         }
     });
 
+    const onClickOverlay = () => {
+        isMobile && hide();
+    };
+
+    const handleKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
+        if (!shouldPreventCloseOnEscape && ev.key === 'Space') {
+            ev.preventDefault();
+            onClickOverlay();
+        }
+    };
+
     return (
-        <div
-            className={classNames('wallets-modal-wrapper', {
-                'wallets-modal-wrapper--fullscreen': isFullscreen,
-            })}
-        >
-            {!hideCloseButton && <CloseIcon className='wallets-modal-wrapper__close-icon' onClick={hide} />}
-            {children}
-        </div>
+        <>
```

Comment:
use Fragment to follow the structure that we have

Reply:
done